packages/client/hmi-client/src/components/code/tera-directory.vue

```diff
@@ -0,0 +1,87 @@
+<template>
+	<aside class="min-w-10rem max-w-20rem">
+		<Tree
+			v-model:selectionKeys="selectedKey"
+			:value="directoryTree"
+			selectionMode="single"
+			:filter="true"
+			filterMode="lenient"
+			filter-placeholder="Find"
+			@node-select="onNodeSelect"
+		>
+		</Tree>
+	</aside>
+</template>
+
+<script setup lang="ts">
+import { onMounted, ref } from 'vue';
+import Tree, { TreeNode, TreeSelectionKeys } from 'primevue/tree';
+
+const directoryTree = ref<TreeNode[]>([]);
+
+const props = defineProps<{
+	files: string[];
+}>();
+
+const emit = defineEmits(['file-clicked']);
+
+const selectedKey = ref<TreeSelectionKeys>();
+onMounted(() => {
+	directoryTree.value = buildDirectoryTree(props.files);
+});
+
+function buildDirectoryTree(files) {
+	const root: TreeNode = {
+		key: '..',
+		label: '..',
+		icon: 'pi pi-fw pi-folder',
+		selectable: false,
+		data: {
+			isDirectory: true
+		},
+		children: []
+	};
+
+	files.forEach((filePath) => {
+		const parts = filePath.split('/');
+		let currentNode: TreeNode = root;
+
+		parts.forEach((part, i) => {
+			const isDirectory = i < parts.length - 1;
+			const existingNode = currentNode.children?.find((node) => node.label === part);
```

Comment:
This `find(...)` is not particularly efficient. Do we have some notion of what is an average-sized code-asset and how deep the paths go?

Reply:
Nope, but those should be done on the back-end eventually, so for now, just to have the feature in is good enough. This should be done once on the repo upload and saved as a JSON or something.

Reply:
I can't say I know how deep the paths go on average...but its a downloaded git repo - so it can be any size.  I'll look to see if I can use a hashmap here to improve the performance instead of using `find(...)`

Reply:
Let's leave it as is then, only Java repos will have very large number of varieties in paths and files - and Java isn't supported by the program currently.